### DISCUSSION TITLE
Missing output file(s) `SRR2962693` expected by process `alignment (SRR2962693)` Error

### DISCUSSION THREAD

**Discussion by hkarakurt8742** [nextflow-io/nextflow#2923] [Q&A]
Hello everyone,
I just started to work with Nextflow and I am trying to build my own workflows step by step. In my first script that I use BWA for alignment. I have the error:

(I use Nextflow DSL 1)

```
Error executing process > 'alignment (SRR2962693)'

Caused by:
  Missing output file(s) `SRR2962693` expected by process `alignment (SRR2962693)`

Command executed:

  bwa mem -M -t 14 /home/huk/Desktop/Tool/references/gatk_19/ref/hg19_v0_Homo_sapiens_assembly19.fasta SRR2962693_1.fastq.gz SRR2962693_2.fastq.gz | samtools sort -@ 14 -o SRR2962693.bam -

Command exit status:
  0

Command output:
  (empty)
```

My script is:

```
/* 
 * pipeline input parameters 
 */
params.reads = "/variant_example/*_{1,2}.fastq.gz"
params.genome_file = "/ref/hg19_v0_Homo_sapiens_assembly19.fasta"
params.outdir = "results"

log.info """\
         ALIGNMENT AND VARIANT CALLING PIPELINE FOR LOCAL USE    
         ===================================
         genome: ${params.genome_file}
         reads        : ${params.reads}
         outdir       : ${params.outdir}
         """
         .stripIndent()

 
/* 
 * define the `index` process that create a binary index 
 * given the transcriptome file
 */


read_pairs_ch = Channel .fromFilePairs(params.reads)

process alignment {
    cpus 14
    tag "$sample_id"
         
    input:
    path genome from params.genome_file
    tuple val(sample_id), path(reads) from read_pairs_ch
 
    output:
    path sample_id into aligned_ch
 
    script:
    """
    bwa mem -M -t $task.cpus ${params.genome_file} ${reads[0]} ${reads[1]} | samtools sort -@ $task.cpus -o ${sample_id}.bam -
    """
}
```

I know it is a really basic question but I could not solve it.

Thank you in advance.

**Comment by bobamess** [comment]
I think your output block in the alignment process should be

```
output:
path “${sample_id}.bam” into aligned_ch
```

**Comment by hkarakurt8742** [comment]
> I think your output block in the alignment process should be
> 
> ```
> output:
> path “${sample_id}.bam” into aligned_ch
> ```

Thank you for you answer. Actually it did not work. I had the error:
       path “${sample_id}.bam” into aligned_ch
            ^

1 error

And when I remove quotes:
Error executing process > 'alignment (SRR2962694)'

Caused by:
  Unknown process directive: `$`

I am not sure what is the real reason of error.

**Reply by bentsherman** [reply]
You definitely need the double quotes. Maybe you also need parentheses?
```
output:
path(“${sample_id}.bam”) into aligned_ch
```

**Comment by bobamess** [comment]
Hmm

Just wondering what the trailing "-" is for in your command line

`bwa mem -M -t $task.cpus ${params.genome_file} ${reads[0]} ${reads[1]} | samtools sort -@ $task.cpus -o ${sample_id}.bam -`

Did you intend to add another option?

And suggest you might also like to add the following line to the end of your script block

`samtools index ${sample_id}.bam`

to get the *.bam.bai file and then also change the output block to

`path sample_id* into aligned_ch`

though I still think it should then be

`path “${sample_id}.bam*” into aligned_ch`

But, maybe there's a difference I'm forgetting about for DSL1 here, as I'm used to using DSL2

**Reply by bentsherman** [reply]
I think the `-` at the end just means that `samtools` will read from stdin, which will be the output of `bwa`.
